### ISSUE TITLE
Provide a synchronous API

### ISSUE THREAD

**Issue by Woodz** [danday74/envsub#3]
Providing an asynchronous API that returns Promises forces an asynchronous approach instead of giving developers the flexibility to use the API in a synchronous or asynchronous way. I propose something like:

```
var result = envsub_sync(templateFile, outputFile, options);
```

**Comment by danday74** [comment]
I totally agree with you. I discovered sync was the best way to offer a module after developing this. I will try and get round to doing this as I am a busy man.

**Comment by danday74** [comment]
Now that this module is more popular, I am willing to put the effort in to fix this. However, I only plan to do so if there is more demand.

I get the impression that envsub is largely used as a global install and thus fixing this issue seems less important.

Does anyone else want this fixed? If so let me know and I will crack on with it.

Leaving open for now.

**Comment by skorunka** [comment]
Me ;)

**Comment by skorunka** [comment]
Can be echieved by using https://stackoverflow.com/a/54139860/522322

			const _envsub = () => envsub({ templateFile, outputFile, options })
				.then((x) => console.log('done'))
				.catch((err) => console.error('[ERROR]', err.message));

			const sync_envsub = sp(_envsub);
			sync_envsub();
